pkg/controller/kibana/controller.go

```diff
@@ -59,20 +117,151 @@ type ReconcileKibana struct {
+
+	// get CR
+	kibanaInstance := &loggingv1.Kibana{}
+	key := types.NamespacedName{
+		Name:      request.Name,
+		Namespace: request.Namespace,
+	}
+
+	err := r.client.Get(context.TODO(), key, kibanaInstance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// the CR no longer exists, since it will be cleaned up by the scheduler we don't want to trigger an event for it
+			unregisterKibanaNamespacedName(request)
+			return reconcile.Result{}, nil
+		}
+
+		return reconcile.Result{}, err
+	}
+
+	if kibanaInstance.Spec.ManagementState == loggingv1.ManagementStateUnmanaged {
+		return reconcile.Result{}, nil
+	}
+
+	// keep track of the fact that we processed this kibana for future events and for mapping
+	registerKibanaNamespacedName(request)
+
 	es, err := k8shandler.GetElasticsearchCR(r.client, request.Namespace)
 	if err != nil {
 		logrus.Infof("skipping kibana reconciliation in %q: %s", request.Namespace, err)
 		return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
 	}
 
 	esClient := elasticsearch.NewClient(es.Name, es.Namespace, r.client)
-	if err := kibana.Reconcile(request, r.client, esClient); err != nil {
-		if errors.IsNotFound(err) {
-			return reconcile.Result{Requeue: true}, nil
-		}
-		logrus.Errorf("kibana reconcile err %v", err)
+
+	proxyCfg, err := kibana.GetProxyConfig(r.client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if err := kibana.Reconcile(kibanaInstance, r.client, esClient, proxyCfg); err != nil {
 		return reconcile.Result{}, err
 	}
 
 	return reconcile.Result{}, nil
 }
+
+// handleSecret returns true if metaname is such that (cr_name) matches or that (cr_name + "-proxy") matches
+func handleSecret(meta metav1.Object) bool {
+
+	// iterate over registeredKibanas that match the namespace
+	namespace := meta.GetNamespace()
+
+	for _, kibana := range registeredKibanas {
+		if kibana.Namespace == namespace {
+			if utils.ContainsString(
+				[]string{kibana.Name, fmt.Sprintf("%s-proxy", kibana.Name)},
+				meta.GetName(),
+			) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// handleConfigMap returns true if metaname is contained in the GlobalProxyList constant
+func handleConfigMap(meta metav1.Object) bool {
+
+	// iterate over registeredKibanas that match the namespace
+	namespace := meta.GetNamespace()
+
+	for _, kibana := range registeredKibanas {
+		if kibana.Namespace == namespace {
+			return utils.ContainsString(constants.ReconcileForGlobalProxyList, meta.GetName())
+		}
+	}
+
+	return false
+}
+
+func registerKibanaNamespacedName(request reconcile.Request) {
+	// check to see if the namespaced name is already registered first
+	found := false
+	for _, kibana := range registeredKibanas {
+		if kibana == request.NamespacedName {
+			found = true
+		}
+	}
+
+	// if not, add it to registeredKibanas
+	if !found {
+		logrus.Infof("Registering %v for future events", request.NamespacedName)
+		registeredKibanas = append(registeredKibanas, request.NamespacedName)
```

Comment:
shouldn't you use a mutex for modifying `registeredKibanas`? This is going to fail `go test -race`

Reply:
i thought that even though we get an event queue it still blocks at the `Reconcile` level so we don't have to worry about race conditions

Reply:
Is that something guaranteed by the operator-sdk? Even if that's the case this will prevent us from testing other vulnerabilities since this will make `go test -race` fail indefinitely right?

Reply:
addressed :)